pandas/core/generic.py

```diff
@@ -8761,24 +8763,297 @@ def _doc_parms(cls):
 """
 
 _cnum_doc = """
+Return cumulative %(desc)s over a DataFrame or Series axis.
+
+Returns a DataFrame or Series of the same size containing the cumulative
+%(desc)s.
 
 Parameters
 ----------
-axis : %(axis_descr)s
+axis : {0 or 'index', 1 or 'columns'}, default 0
+    The index or the name of the axis. 0 is equivalent to None or 'index'.
 skipna : boolean, default True
     Exclude NA/null values. If an entire row/column is NA, the result
-    will be NA
+    will be NA.
+*args, **kwargs :
+    Additional keywords have no effect but might be accepted for
+    compatibility with NumPy.
 
 Returns
 -------
-%(outname)s : %(name1)s\n
-
-
+%(outname)s : %(name1)s or %(name2)s\n
+%(examples)s
 See also
 --------
 pandas.core.window.Expanding.%(accum_func_name)s : Similar functionality
     but ignores ``NaN`` values.
+Series.%(outname)s : Return cumulative %(desc)s over Series axis.
+%(name2)s.%(accum_func_name)s : Return the %(desc)s over
+    %(name2)s axis.
+DataFrame.cummax : Return cumulative maximum over DataFrame axis.
```

Comment:
This also `%(name2)s` like below ?

Reply:
If I implement these changes, we won't get a reference to any corresponding `Series` in 'See also' of `DataFrame` methods and vice versa. That's why I left them like this (at the expense of repeating one of them). 

I agree that we don't really need them because we have examples of both `Series` & `DataFrame` in all docstrings and this should be informative enough.

Just out of curiosity, is there a simple way to determine whether a `Series` or `DataFrame` method is being accessed at any one time? For example, if we were generating the doc for `Series.cummax`, we would determine it is a `Series` method. Based on this, we know that we need to show cummax but for a `DataFrame`.

Reply:
The thing is that the docstring for both Series and DataFrame (of the same method) will be exactly the same (apart from some of the links here in see also). So linking to the same method but on the other object, is not that important I think, as the other one does not give you more information

> Just out of curiosity, is there a simple way to determine whether a Series or DataFrame method is being accessed at any one time? For example, if we were generating the doc for Series.cummax, we would determine it is a Series method. Based on this, we know that we need to show cummax but for a DataFrame.

Yes, the `cls` that is passed to `_make_cum_function` is either the Series or DataFrame class. So you can pass `cls.__name__` as a variable to the template, but I think this is already done as `name2` ? Based on that you can know the other.